setup.sh

```diff
@@ -656,20 +655,26 @@ echo "    (ps axf | grep \"webmin\/miniserv\.pl\" | awk '{print \"kill -9 -- -\"
 echo "  fi" >>$config_dir/.stop-init
 echo "fi" >>$config_dir/.stop-init
 # Restart main
-echo "#!/bin/sh" >>$config_dir/.restart-init
+echo "#!/bin/sh" >$config_dir/.restart-init
 echo "$config_dir/.stop-init" >>$config_dir/.restart-init
 echo "$config_dir/.start-init" >>$config_dir/.restart-init
 # Force reload main
-echo "#!/bin/sh" >>$config_dir/.restart-by-force-kill-init
+echo "#!/bin/sh" >$config_dir/.restart-by-force-kill-init
 echo "$config_dir/.stop-init --kill" >>$config_dir/.restart-by-force-kill-init
 echo "$config_dir/.start-init" >>$config_dir/.restart-by-force-kill-init
 # Reload main
-echo "#!/bin/sh" >>$config_dir/.reload-init
+echo "#!/bin/sh" >$config_dir/.reload-init
 echo "echo Reloading Webmin server in $wadir" >>$config_dir/.reload-init
 echo "pidfile=\`grep \"^pidfile=\" $config_dir/miniserv.conf | sed -e 's/pidfile=//g'\`" >>$config_dir/.reload-init
 echo "kill -USR1 \`cat \$pidfile\`" >>$config_dir/.reload-init
-
-chmod 755 $config_dir/.stop-init $config_dir/.start-init $config_dir/.restart-init $config_dir/.restart-by-force-kill-init $config_dir/.reload-init
+# Pre install
+echo "#!/bin/sh" >$config_dir/.pre-install
+echo "$config_dir/.stop-init" >>$config_dir/.pre-install
+# Post install
+echo "#!/bin/sh" >$config_dir/.post-install
+echo "$config_dir/.start-init" >>$config_dir/.post-install
```

Comment:
Why does .post-install call .start-init here, when in setup.pl it sends a HUP signal below on line 726?

Reply:
This code (620-675) is for the older systems only (without `systemd`), while code on lines 700-726 is only for new systems, therefor new systems (with `systemd`) will have `SIGHUP` line, while older systems will call initial start script.

This behavior replicates old logic, where at first on older systems you called `/etc/webmin/stop` (and now I call `SIGSTOP` on new systems with `systemd`) and after setup you called `/etc/webmin/start` (and I call `SIGCONT` and `SIGHUP` on newer systems).

That said, the code in files `/etc/webmin/.pre-install` and `/etc/webmin/.post-install` on old systems and new systems won't be the same.

Reply:
Ok, I think I see. As I mentioned, the only change I'd recommend is remove the STOP/CONT signals.

Reply:
Removed for now.